src.ts/contract/contract.ts

```diff
@@ -317,6 +321,17 @@ function buildWrappedMethod<A extends Array<any> = Array<any>, R = any, D extend
         return new ContractTransactionResponse(contract.interface, <Provider>provider, tx);
     }
 
+    const delegateCall = async function(...args: ContractMethodArgs<A>): Promise<ContractTransactionResponse> {
+        const fragment = getFragment(...args);
+        // If an overrides was passed in, copy it and normalize the values
+        if (fragment.inputs.length + 1 === args.length) {
+            Object.assign(args[args.length - 1], { type: 4 });
```

Comment:
Current implementation "leaks" `types: 4` into the args.at(-1) ... that may be reused elsewhere.
```suggestion
            args.at(-1) = Object.assign({}, args.at(-1), { type: 4 });
```